### PR TITLE
XW-4645 | Implement Mobile Site link on Oasis for mobile

### PR DESCRIPTION
**Pull request by kolek7**
@Wikia/x-wing 

https://wikia-inc.atlassian.net/browse/XW-4645

### CODE REVIEW

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require(['jquery'],
+	function ($) {
+		if ($.cookie('useskin') === 'oasis')  {
+			$('.mobile-site-link').click(mobileSiteLinkClicked);
+			$('.mobile-site-link').css("display", "inline-block");
```
Comment (rybmat) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js]: do we need to do it via JS? what about simple css like this:
`.WikiaSiteWrapper .wds-global-footer .mobile-site-link {display: block;}` ?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 2]
```diff
@@ -212,6 +212,13 @@ public function getData() {
 					]
 				],
 			],
+			'mobile_site_button' => [
```
Comment (rybmat) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:215]: during QA it is worth to test if it won't break mobile-wiki, discussions-frontend or fandom

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require(['jquery'],
+	function ($) {
+		if ($.cookie('useskin') === 'oasis')  {
```
Comment (rybmat) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js:3]: is this mobile-link button intended to show only if user clicked the full site link on mobile wiki? AFAIK this cookie is set only in this case. Maybe we could show it always on desktop?

**Review by rybmat — CHANGES_REQUESTED** [review]

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js (hunk 3 above)
Comment (bkoval) [reply extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js:3]: Nah, I think that's what we want in the ticket - to only show it when you switch to oasis on a mobile device.

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+			$('.mobile-site-link').click(mobileSiteLinkClicked);
```
Comment (bkoval) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js]: You can chain these, e.g.
```
$('.mobile-site-link')
    .click(mobileSiteLinkClicked)
    .css("display", "inline-block");
```
This way you do not evaluate the same selector twice (makes performance better, all the DOM operations are performance-costly).

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js (hunk 1 above)
Comment (bkoval) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalFooter.js]: We need to set it dynamically as we only show it when the cookie is set. That set, we can prepare a class in .scss, e.g. "mobile-site-link--active" that will set display: inline-block and we assign this class here, e.g.
```
.addClass ('mobile-site-link--active');
```

This way we keep all the presentation (styling) layer in the SCSS file :)

File: extensions/wikia/DesignSystem/services/templates/DesignSystemGlobalFooter_licensingAndVertical.php [hunk 5]
```diff
@@ -1,3 +1 @@
-<div class="wds-global-footer__bottom-bar">
-	<div class="wds-global-footer__bottom-bar-row wds-has-padding"><?= DesignSystemHelper::renderText( $model['description'] ) ?></div>
-</div>
+<div class="wds-global-footer__bottom-bar-row wds-has-padding"><?= DesignSystemHelper::renderText( $model['description'] ) ?></div>
```
Comment (bkoval) [inline extensions/wikia/DesignSystem/services/templates/DesignSystemGlobalFooter_licensingAndVertical.php:1]: Why did you remove the wrapper div?

File: extensions/wikia/DesignSystem/styles/DesignSystemGlobalFooter.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.mobile-site-link {
+  display: none;
```
Comment (bkoval) [inline extensions/wikia/DesignSystem/styles/DesignSystemGlobalFooter.scss]: Indentation should use one tab, not two spaces :)

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 7]
```diff
@@ -212,6 +212,13 @@ public function getData() {
...
+				'type' => 'link-text',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-footer-mobile-site-link'
```
Comment (bkoval) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:219]: Was that already translated?

**Review by bkoval — CHANGES_REQUESTED** [review]
Good job! Left a couple of comments.

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php (hunk 7 above)
Comment (kolek7) [reply includes/wikia/models/DesignSystemGlobalFooterModel.class.php:219]: Fortunately, yes 🙂

File: extensions/wikia/DesignSystem/styles/DesignSystemGlobalFooter.scss (hunk 6 above)
Comment (kolek7) [reply extensions/wikia/DesignSystem/styles/DesignSystemGlobalFooter.scss]: Gotta configure my IDE config then.

File: extensions/wikia/DesignSystem/services/templates/DesignSystemGlobalFooter_licensingAndVertical.php (hunk 5 above)
Comment (kolek7) [reply extensions/wikia/DesignSystem/services/templates/DesignSystemGlobalFooter_licensingAndVertical.php:1]: I moved this wrapper to `DesignSystemGlobalFooter_index.php`, because I wanted to nest both licensingAndVertical template and mobileSiteButton template inside it.

File: extensions/wikia/DesignSystem/services/DesignSystemGlobalFooterService.class.php [hunk 8]
```diff
@@ -29,6 +29,10 @@ public function licensingAndVertical() {
 		$this->setVal( 'model', $this->getVal( 'model' ) );
 	}
 
+	public function mobileSiteButton() {
+		$this->setVal( 'model', $this->getVal( 'model' ) );
```
Comment (mszabo-wikia) [inline extensions/wikia/DesignSystem/services/DesignSystemGlobalFooterService.class.php]: you could use `renderPartial` instead when rendering this template, it wont invoke the full dispatcher process and doesn't require having such a boilerplate method defined

**Review by rybmat — APPROVED** [review]

**Review by JamesHawking — APPROVED** [review]

**Review by bkoval — APPROVED** [review]
